etcdctlv3/command/txn_command.go

```diff
@@ -41,198 +51,178 @@ func txnCommandFunc(cmd *cobra.Command, args []string) {
-					Key: key,
-				}}
-		}
-	case "p", "put":
-		ru.Request = &pb.RequestUnion_RequestPut{
-			RequestPut: &pb.PutRequest{
-				Key:   key,
-				Value: []byte(parts[2]),
-			}}
-	case "d", "deleteRange":
-		if len(parts) == 3 {
-			ru.Request = &pb.RequestUnion_RequestDeleteRange{
-				RequestDeleteRange: &pb.DeleteRangeRequest{
-					Key:      key,
-					RangeEnd: []byte(parts[2]),
-				}}
-		} else {
-			ru.Request = &pb.RequestUnion_RequestDeleteRange{
-				RequestDeleteRange: &pb.DeleteRangeRequest{
-					Key: key,
-				}}
-		}
+	cmpType := ""
+	switch parts[2] {
+	case "g", "greater":
+		cmpType = ">"
+	case "e", "equal":
+		cmpType = "="
+	case "l", "less":
+		cmpType = "<"
 	default:
-		return nil, fmt.Errorf("invalid txn request: %s", line)
-	}
-	return ru, nil
-}
-
-func parseCompare(line string) (*pb.Compare, error) {
-	parts := strings.Split(line, " ")
-	if len(parts) != 4 {
 		return nil, fmt.Errorf("invalid txn compare request: %s", line)
 	}
 
-	var err error
-	c := &pb.Compare{}
-	c.Key = []byte(parts[0])
+	var (
+		v   int64
+		err error
+		cmp clientv3.Cmp
+	)
+
+	key := parts[0]
 	switch parts[1] {
 	case "ver", "version":
-		tv, _ := c.TargetUnion.(*pb.Compare_Version)
-		if tv != nil {
-			tv.Version, err = strconv.ParseInt(parts[3], 10, 64)
-			if err != nil {
-				return nil, fmt.Errorf("invalid txn compare request: %s", line)
-			}
+		if v, err = strconv.ParseInt(parts[3], 10, 64); err != nil {
+			cmp = clientv3.Compare(clientv3.Version(key), cmpType, v)
 		}
 	case "c", "create":
-		tv, _ := c.TargetUnion.(*pb.Compare_CreateRevision)
-		if tv != nil {
-			tv.CreateRevision, err = strconv.ParseInt(parts[3], 10, 64)
-			if err != nil {
-				return nil, fmt.Errorf("invalid txn compare request: %s", line)
-			}
+		if v, err = strconv.ParseInt(parts[3], 10, 64); err != nil {
+			cmp = clientv3.Compare(clientv3.CreatedRevision(key), cmpType, v)
 		}
 	case "m", "mod":
-		tv, _ := c.TargetUnion.(*pb.Compare_ModRevision)
-		if tv != nil {
-			tv.ModRevision, err = strconv.ParseInt(parts[3], 10, 64)
-			if err != nil {
-				return nil, fmt.Errorf("invalid txn compare request: %s", line)
-			}
+		if v, err = strconv.ParseInt(parts[3], 10, 64); err != nil {
+			cmp = clientv3.Compare(clientv3.ModifiedRevision(key), cmpType, v)
 		}
 	case "val", "value":
-		tv, _ := c.TargetUnion.(*pb.Compare_Value)
-		if tv != nil {
-			tv.Value = []byte(parts[3])
-		}
-	default:
-		return nil, fmt.Errorf("invalid txn compare request: %s", line)
+		cmp = clientv3.Compare(clientv3.Value(key), cmpType, parts[3])
 	}
 
-	switch parts[2] {
-	case "g", "greater":
-		c.Result = pb.Compare_GREATER
-	case "e", "equal":
-		c.Result = pb.Compare_EQUAL
-	case "l", "less":
-		c.Result = pb.Compare_LESS
-	default:
+	if err != nil {
 		return nil, fmt.Errorf("invalid txn compare request: %s", line)
 	}
-	return c, nil
+
+	return &cmp, nil
+}
+
+func printTxnResponse(resp clientv3.TxnResponse, isHex bool) {
+	if resp.Succeeded {
+		fmt.Println("SUCCESS")
```

Comment:
we use \r\n to do the separation in other commands. should we adopt that format?